src/sqlite.h.in

```diff
@@ -10365,6 +10365,22 @@ int sqlite3_preupdate_new(sqlite3 *, int, sqlite3_value **);
 int sqlite3_preupdate_blobwrite(sqlite3 *);
 #endif
 
+/*
+** CAPI3REF: The close hook.
+** METHOD: sqlite3
+**
+** ^The [libsql_close_hook()] interface registers a callback function
+** that is invoked prior to closing the database connection.
+** ^At most one close hook may be registered at a time on a single
+** [database connection]; each call to [libsql_close_hook()] overrides
+** the previous setting.
+** ^The close hook is disabled by invoking [libsql_close_hook()]
+** with a NULL pointer as the second parameter.
+** ^The third parameter to [libsql_close_hook()] is passed through as
+** the first parameter to callbacks.
+*/
```

Comment:
Let's mention somewhere that this is called with sqlite3 object mutex held so people know they can't call the sqlite API.

Reply:
Oh but they can and most likely will. The main db mutex is recursive so it's fine to reenter it from the same fiber: https://github.com/libsql/libsql/blob/c2cf4380bbf254ec7bd2f10c025a260a123bdcb5/src/main.c#L3296